fix value popups displaying under other elements

bekand opened the pull request:
# Terms

> THIS SOFTWARE IS CONTRIBUTED SUBJECT TO THE TERMS OF THE TERMS OF THE CCLA DATED 2017-11-07 WITH FINOS/LINUX FOUNDATION (FORMERLY THE SYMPHONY SOFTWARE FOUNDATION CCLA).
>
> THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.

AttilaMihaly commented on cli/src/Morphir/Web/DevelopApp.elm:
```diff
@@ -2135,7 +2135,7 @@ viewDefinitionDetails model =
             case fromRawValue ir rawValue of
                 Ok typedValue ->
-                    el [ centerY ] (ViewValue.viewValue config typedValue)
+                    el [ centerY ] (ViewValue.viewValue 1000 config typedValue)
```
It would be good to move the specific numbers out to named variables and also add some explanation on why the particular value was chosen. I see 1000 and 9999 but not sure why those are the values.

AttilaMihaly commented on cli/src/Morphir/Web/DevelopApp.elm:
```diff
@@ -2315,7 +2315,7 @@ viewDefinitionDetails model =
                                                                                 , onToggle = UI (ToggleSection 1)
                                                                                 , isOpen = Set.member 1 model.openSections
-                                                                                , content = el [ Theme.borderRounded model.theme, Border.width 1, Border.color model.theme.colors.gray ] <| ViewValue.viewDefinition (insightViewConfig ir) fullyQualifiedName valueDef
+                                                                                , content = el [ Theme.borderRounded model.theme, Border.width 1, Border.color model.theme.colors.gray] <| ViewValue.viewDefinition 9999 (insightViewConfig ir) fullyQualifiedName valueDef
```
Same here. Move to named value/constant.

AttilaMihaly commented on src/Morphir/Visual/ViewApply.elm:
```diff
@@ -42,12 +51,36 @@ view config viewDefinitionBody viewValue functionValue argValues =
+
+                            _ ->
+                                el
+                                    [ htmlAttribute (style "filter" "brightness(96%)")
```
Can we simply use a background color here instead of a filter? In case of a drill-down the brightness filter serves the specific purpose of showing depth. We don't need that for arguments. A background with a low alpha would be much simpler.

AttilaMihaly commented on src/Morphir/Visual/ViewArithmetic.elm:
```diff
@@ -9,8 +9,8 @@ import Morphir.Visual.EnrichedValue exposing (EnrichedValue)
-view : Config msg -> (EnrichedValue -> Element msg) -> ArithmeticOperatorTree -> Element msg
-view config viewValue arithmeticOperatorTree =
+view : Config msg -> Int -> (Int -> EnrichedValue -> Element msg) -> ArithmeticOperatorTree -> Element msg
+view config zIndex viewValue arithmeticOperatorTree =
```
Would it be possible to include the `zIndex` in the config or derive it from the `EnrichedValue` somehow? Our goal here is to have different `zIndex` values for each drill-down level. Sounds like that information is already available given that we are tracking expanded/collapsed state per level.

bekand replied:
we don't just want to have drilldowns have lower z-index thank their parents, unfortunately they have to have a lover z-index than previous drilldowns on the same level of the expression, because of cases like the attached image
![image](https://github.com/finos/morphir-elm/assets/18498914/ba0cd5e3-3239-4099-9618-865ffa9910b6)

AttilaMihaly commented on src/Morphir/Visual/ViewValue.elm:
```diff
@@ -481,4 +483,5 @@ viewDrillDown config value fQName letDefOpenElement =
         , openElement = openElement
         , openHeader = closedElement
         , isOpen = drillDownContains config.state.drillDownFunctions id config.nodePath
+        , zIndex = zIndex - 100
```
Why is this decremented by 100 while it's decremented by 1 everywhere else?

bekand commented:
I moved the z-index into the `VisualState` part of the config, and changed the variable backgrounds from filter to simple background, so we have to deal with the zIndex in less places now
